### PR TITLE
DS routing/serialization

### PR DESCRIPTION
**Pull request by rebstar6**
should be 2nd to last PR; get that serialization!

### CODE REVIEW

File: modules/api/functional_test/live_tests/shared_zone_test_context.py [hunk 1]
```diff
@@ -243,6 +243,29 @@ def __init__(self):
...
+                    },
+                    'transferConnection': {
+                        'name': 'example.',
+                        'keyName': VinylDNSTestContext.dns_key_name,
+                        'key': VinylDNSTestContext.dns_key,
+                        'primaryServer': VinylDNSTestContext.dns_ip
+                    }
+                }, status=202)
+            self.ds_zone = ds_zone_change['zone']
+
```
Comment (nimaeskandary) [inline modules/api/functional_test/live_tests/shared_zone_test_context.py]: should we have the extra validation on line 275? `wait until our zones are created`

File: modules/api/src/test/scala/vinyldns/api/route/VinylDNSJsonProtocolSpec.scala [hunk 2]
```diff
@@ -269,15 +270,15 @@ class VinylDNSJsonProtocolSpec
     "parse a record set with an absolute CNAME record passes" in {
       val recordSetJValue: JValue =
         ("zoneId" -> "1") ~~
-          ("name" -> "Comcast") ~~
```
Comment (nimaeskandary) [inline modules/api/src/test/scala/vinyldns/api/route/VinylDNSJsonProtocolSpec.scala]: 🦚

File: modules/core/src/main/scala/vinyldns/core/domain/record/RecordData.scala [hunk 3]
```diff
@@ -102,15 +102,33 @@ object DnsSecAlgorithm {
...
+  case object PRIVATEOID extends DnsSecAlgorithm(254)
   final case class UnknownAlgorithm private (x: Int) extends DnsSecAlgorithm(x)
 
   def apply(value: Int): DnsSecAlgorithm =
     value match {
       case 3 => DSA
       case 5 => RSASHA1
       case 6 => DSA_NSEC3_SHA1
-      case 7 => RSA_NSEC3_SHA1
+      case 7 => RSASHA1_NSEC3_SHA1
```
Comment (nimaeskandary) [inline modules/core/src/main/scala/vinyldns/core/domain/record/RecordData.scala]: Checked against https://www.iana.org/assignments/dns-sec-alg-numbers/dns-sec-alg-numbers.xhtml and all look right

File: modules/api/functional_test/live_tests/shared_zone_test_context.py (hunk 1 above)
Comment (rebstar6) [reply modules/api/functional_test/live_tests/shared_zone_test_context.py]: added

File: modules/api/src/main/scala/vinyldns/api/route/DnsJsonProtocol.scala [hunk 4]
```diff
@@ -435,6 +438,38 @@ trait DnsJsonProtocol extends JsonValidation {
         ("fingerprint" -> rr.fingerprint)
   }
 
+  case object DSSerializer extends ValidationSerializer[DSData] {
+    override def fromJson(js: JValue): ValidatedNel[String, DSData] =
+      (
+        (js \ "keytag")
```
Comment (mitruly) [inline modules/api/src/main/scala/vinyldns/api/route/DnsJsonProtocol.scala]: Should probably use camel-casing here for consistency (`keyTag` and `digestType`).

Comment (mitruly) [reply modules/api/src/main/scala/vinyldns/api/route/DnsJsonProtocol.scala]: Actually nvm, this is for the actual DNS record data. Lowercase all the way~

**Pull request merged**